packages/components/psammead-navigation/hooks/useMediaQuery.test.js

```diff
@@ -0,0 +1,17 @@
+import { renderHook } from '@testing-library/react-hooks';
+import useMediaQuery from './useMediaQuery';
+
+describe('MediaQuery hook', () => {
+  it('it should test that the media query matches', async () => {
+    window.matchMedia = jest.fn().mockImplementation(query => {
+      return {
+        matches: query === '(max-width: 600px)',
+        addListener: jest.fn(),
+        removeListener: jest.fn(),
+      };
+    });
+
+    const { result } = renderHook(() => useMediaQuery('(max-width: 600px)'));
+    expect(result.current).toBe(true);
+  });
+});
```

Comment:
This seems to not test the `useMediaQuery` hook. If I delete the `useEffect` block in `useMediaQuery` hook the test still pass.  This test seem to test `useState given an initial state it will return the set state`. For this test to be complete we would need to find a way to trigger a window resize to make use of the `useEffect` block.

Reply:
You are right. Since I didn't find a way to trigger the window resize from the test, what I have done is to put 600px in the mock, so when I call the `useMediaQuery` hook with `(max-width: 600px)` I expect it to return true.

Reply:
We can check that the `addListener` function has been called. We may also be able to check the effects of the function are what we expect. Looking into that as well.

```jsx
    const mockAddListener = jest.fn();
    const mockRemoveListener = jest.fn();

    window.matchMedia = jest.fn().mockImplementation(query => {
      return {
        matches: query === '(max-width: 600px)',
        addListener: mockAddListener,
        removeListener: mockRemoveListener,
      };
    });

    const { result } = renderHook(() => useMediaQuery('(max-width: 600px)'));
    expect(result.current).toBe(true);
    expect(window.matchMedia.addListener).toBeCalled();
```

Reply:
Here's a version of test that also checks the `addListener` function updates the `result.current`. Note that this should probably be at least two separate tests.

```jsx
  it('it should test that the media query matches', async () => {
    const mockAddListener = jest.fn();
    const mockRemoveListener = jest.fn();

    window.matchMedia = jest.fn().mockImplementation(query => {
      return {
        matches: query === '(max-width: 600px)',
        addListener: mockAddListener,
        removeListener: mockRemoveListener,
      };
    });

    const { result } = renderHook(() => useMediaQuery('(max-width: 600px)'));
    expect(result.current).toBe(true);
    expect(mockAddListener).toBeCalled();

    act(() => mockAddListener.mock.calls[0][0]({ matches: false }));
    expect(result.current).toBe(false);
    act(() => mockAddListener.mock.calls[0][0]({ matches: true }));
    expect(result.current).toBe(true);
  }
```

Reply:
Done. Thanks for the suggestions :)